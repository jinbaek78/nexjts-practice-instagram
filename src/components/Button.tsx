type Props = {
  text: string;
  textSize: string;
};
const BUTTON_CLASS =
  'rounded-md bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 p-1';
export default function Button({ text, textSize = 'text-xl' }: Props) {
  return (
    <div className={`${BUTTON_CLASS} ${textSize}`}>
      <button className="p-2 h-full w-full bg-gray-50 font-medium ">
        {text}
      </button>
    </div>
  );
}
