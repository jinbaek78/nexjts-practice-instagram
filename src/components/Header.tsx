'use client';

import { AiOutlineHome, AiFillHome } from 'react-icons/ai';
import { BiSearch, BiSolidSearchAlt2 } from 'react-icons/bi';
import { BsPlusSquare, BsPlusSquareFill } from 'react-icons/bs';
import Button from './Button';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { useSession } from 'next-auth/react';

const ICON_CLASS = 'mr-6';
export default function Header() {
  const { data: session } = useSession();
  const buttonText = session ? 'Sign out' : 'Sign in';
  const pathname = usePathname();
  return (
    <header className="flex justify-between my-2  p-2 max-w-screen-2xl mx-auto ">
      <section className="flex flex-col justify-center items-center">
        <Link href={'/'}>
          <h1 className="font-bold text-4xl">Instagram</h1>
        </Link>
      </section>
      <nav className="flex justify-evenly items-center text-3xl">
        <Link href={'/'} className={ICON_CLASS}>
          {pathname === '/' ? <AiFillHome /> : <AiOutlineHome />}
        </Link>

        <Link href={'/search'} className={ICON_CLASS}>
          {pathname === '/search' ? <BiSolidSearchAlt2 /> : <BiSearch />}
        </Link>

        <Link href={'/new'} className={ICON_CLASS}>
          {pathname === '/new' ? <BsPlusSquareFill /> : <BsPlusSquare />}
        </Link>
        <div className={ICON_CLASS}>
          <Button text={buttonText} />
        </div>
      </nav>
    </header>
  );
}
