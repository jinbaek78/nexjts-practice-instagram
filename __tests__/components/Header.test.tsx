import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '@/components/Header';
import mockRouter from 'next-router-mock';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import Button from '@/components/Button';

jest.mock('@/components/Button');

describe('Header', () => {
  afterEach(() => {
    (Button as jest.Mock).mockReset();
  });

  it('navigates to home page on instagram title click', () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    const links = screen.getAllByRole('link');
    fireEvent.click(links[0]);

    expect(mockRouter.asPath).toBe('/');
  });

  it('navigates to home page when home icon is clicked', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    const links = screen.getAllByRole('link');
    await fireEvent.click(links[1]);

    expect(mockRouter.asPath).toBe('/');
  });

  it('navigates to new page when search icon is clicked', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    const links = screen.getAllByRole('link');
    await fireEvent.click(links[2]);

    expect(mockRouter.asPath).toBe('/search');
  });

  it('navigates to new page when search icon is clicked', () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    const links = screen.getAllByRole('link');
    fireEvent.click(links[2]);

    expect(mockRouter.asPath).toBe('/search');
  });

  it('should call button component with text and textsize arguments on button click', () => {
    render(<Header />);
    expect(Button).toHaveBeenCalledTimes(1);
    expect((Button as jest.Mock).mock.calls[0][0].text).toBe('Sign in');
    expect((Button as jest.Mock).mock.calls[0][0].textSize).toBe('text-xl');
  });
});
