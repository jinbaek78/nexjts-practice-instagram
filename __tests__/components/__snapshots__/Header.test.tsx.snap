// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should re-render 1`] = `
<DocumentFragment>
  <header
    class="flex justify-between my-2  p-2 max-w-screen-2xl mx-auto "
  >
    <section
      class="flex flex-col justify-center items-center"
    >
      <a
        href="/"
      >
        <h1
          class="font-bold text-4xl"
        >
          Instagram
        </h1>
      </a>
    </section>
    <nav
      class="flex justify-evenly items-center text-3xl"
    >
      <a
        class="mr-6"
        href="/"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
          />
        </svg>
      </a>
      <a
        class="mr-6"
        href="/search"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
          />
        </svg>
      </a>
      <a
        class="mr-6"
        href="/new"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
          />
          <path
            d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
          />
        </svg>
      </a>
      <div
        class="mr-6"
      >
        <div
          class="rounded-md bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 p-1 text-xl"
        >
          <button
            class="p-2 h-full w-full bg-gray-50 font-medium "
          >
            Sign in
          </button>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`Header should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="flex justify-between my-2  p-2 max-w-screen-2xl mx-auto "
      >
        <section
          class="flex flex-col justify-center items-center"
        >
          <a
            href="/"
          >
            <h1
              class="font-bold text-4xl"
            >
              Instagram
            </h1>
          </a>
        </section>
        <nav
          class="flex justify-evenly items-center text-3xl"
        >
          <a
            class="mr-6"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </a>
          <a
            class="mr-6"
            href="/search"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
              />
            </svg>
          </a>
          <a
            class="mr-6"
            href="/new"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
              />
              <path
                d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
              />
            </svg>
          </a>
          <div
            class="mr-6"
          >
            <div
              class="rounded-md bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 p-1 text-xl"
            >
              <button
                class="p-2 h-full w-full bg-gray-50 font-medium "
              >
                Sign in
              </button>
            </div>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="flex justify-between my-2  p-2 max-w-screen-2xl mx-auto "
    >
      <section
        class="flex flex-col justify-center items-center"
      >
        <a
          href="/"
        >
          <h1
            class="font-bold text-4xl"
          >
            Instagram
          </h1>
        </a>
      </section>
      <nav
        class="flex justify-evenly items-center text-3xl"
      >
        <a
          class="mr-6"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
            />
          </svg>
        </a>
        <a
          class="mr-6"
          href="/search"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
          </svg>
        </a>
        <a
          class="mr-6"
          href="/new"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
            />
            <path
              d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
            />
          </svg>
        </a>
        <div
          class="mr-6"
        >
          <div
            class="rounded-md bg-gradient-to-r from-pink-500 via-red-500 to-yellow-500 p-1 text-xl"
          >
            <button
              class="p-2 h-full w-full bg-gray-50 font-medium "
            >
              Sign in
            </button>
          </div>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
